### PR TITLE
modernize to pyproject.toml

### PR DESCRIPTION
**Pull request by mgeplf**

### CODE REVIEW

**Review by GianlucaFicarelli — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,64 @@
...
 requires = [
-    "setuptools>=42",
-    "wheel",
+    "setuptools>=64",
+    "setuptools_scm>=8"
 ]
 build-backend = "setuptools.build_meta"
+
+[project]
+name = "neurom"
```
Comment (eleftherioszisis) [inline pyproject.toml:9]: Doesn't that have to be NeuroM. It's project name, not package name. Right?

Comment (mgeplf) [reply pyproject.toml]: not sure; I've set it to `NeuroM` now.

Comment (mgeplf) [reply pyproject.toml]: Note that the original has:
``
name='neurom',
```

Comment (mgeplf) [reply pyproject.toml:9]: I guess we'll go back to the original

**Pull request merged**